### ISSUE TITLE
3.0.107/FUSE3: FTBFS on i686 :: "fuse: off_t must be 64bit"

### ISSUE THREAD

**Issue by onlyjob** [moosefs/moosefs#311]
3.0.107 FTBFS with libfuse3-3.7.0 on _i686_ as follows:

```
In file included from /usr/include/fuse3/fuse_lowlevel.h:25,
                 from fusecommon.h:14,
                 from mfs_fuse.h:24,
                 from extrapackets.c:29:
/usr/include/fuse3/fuse_common.h:817:1: error: static assertion failed: "fuse: off_t must be 64bit"
  817 | _Static_assert(sizeof(off_t) == 8, "fuse: off_t must be 64bit");
      | ^~~~~~~~~~~~~~
```

**Comment by acid-maker** [comment]
It should be fixed in libfuse3 not in Moosefs. Type `off_t` on 32-bit Linux is by default 32-bit and you have to add -D_FILE_OFFSET_BITS=64 to change `off_t` to 64-bit. Since this define is essential for fuse3 it should be added to pkgconfig file distributed with package libfuse3-dev.

Of course it is not harmful to add it to default Linux defines in MooseFS and therefore I'll add it, but consider contacting with libfuse3-dev maintainer and telling him/her to add this to pkgconfig file.

**Comment by acid-maker** [comment]
fixed by commit 9299480caabe37ef51e8c349e2ea94ee2d339295

**Comment by onlyjob** [comment]
Thanks! Situation is much improved with 9299480.
Unfortunately upstream was unwilling to address the problem...
